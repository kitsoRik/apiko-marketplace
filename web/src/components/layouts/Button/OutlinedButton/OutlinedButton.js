import React from "react";

import "./OutlinedButton.scss";

const OutlinedButton = ({
	className,
	uppercase = false,
	value,
	icon,
	...props
}) => {
	return (
		<button
			className={`button-outlined ${className ?? ""}`}
			style={{ textTransform: uppercase ? "uppercase" : "none" }}
			{...props}
		>
			{icon && <div className="button-outlined-icon">{icon}</div>}
			{value}
		</button>
	);
};

export default OutlinedButton;
