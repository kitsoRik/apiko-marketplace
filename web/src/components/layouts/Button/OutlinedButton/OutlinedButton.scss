.button-outlined {
	padding: 0px 7px;

	display: flex;
	justify-content: center;
	align-items: center;

	color: #535353;

	background-color: white;

	border: 2px solid #349a89;
	border-radius: 5px;

	box-sizing: border-box;

	&:hover {
		color: black;
		background-color: #ddd;
	}

	&-icon {
		margin-right: 14px;
	}
}
