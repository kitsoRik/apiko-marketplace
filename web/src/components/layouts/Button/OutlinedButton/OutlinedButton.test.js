import React from "react";
import { create } from "react-test-renderer";
import OutlinedButton from "./OutlinedButton";

describe("OutlinedButton", () => {
	it("should render without value", () => {
		const component = create(<OutlinedButton />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with value", () => {
		const component = create(<OutlinedButton value="My custom value" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with className", () => {
		const component = create(
			<OutlinedButton className="my-custom-class-name" />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
