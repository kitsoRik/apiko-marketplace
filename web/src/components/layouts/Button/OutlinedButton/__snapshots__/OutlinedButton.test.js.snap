// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedButton should render with className 1`] = `
<button
  className="button-outlined my-custom-class-name"
  style={
    Object {
      "textTransform": "none",
    }
  }
/>
`;

exports[`OutlinedButton should render with value 1`] = `
<button
  className="button-outlined "
  style={
    Object {
      "textTransform": "none",
    }
  }
>
  My custom value
</button>
`;

exports[`OutlinedButton should render without value 1`] = `
<button
  className="button-outlined "
  style={
    Object {
      "textTransform": "none",
    }
  }
/>
`;
